js/data/bucket/fill_bucket.js

```diff
@@ -60,14 +60,12 @@ FillBucket.prototype.addFeature = function(feature) {
     var lines = loadGeometry(feature);
     var polygons = classifyRings(lines, EARCUT_MAX_RINGS);
 
-    var startGroup = this.prepareArrayGroup('fill', 0);
-    var startIndex = startGroup.layoutVertexArray.length;
-
+    var startVertex = this.vertexArrayLength('fill');
     for (var i = 0; i < polygons.length; i++) {
         this.addPolygon(polygons[i]);
     }
-
-    this.populatePaintArrays('fill', {zoom: this.zoom}, feature.properties, startGroup, startIndex);
+    var endVertex = this.vertexArrayLength('fill') - 1;
```

Comment:
Looks like we need to do `endVertex - 1` everywhere. Does this have to do with `Bucket#populatePaintArrays`'s `endIndex` being exclusive?